src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -201,22 +231,35 @@ private static RedirectionItem[] LoadRedirectionModel(ErrorBuilder errors, strin
                             });
                         }
                     }
-
-                    return results.OrderBy(item => item.RedirectUrl.Source).ToArray();
                 }
             }
 
-            return Array.Empty<RedirectionItem>();
+            return results.OrderBy(item => item.RedirectUrl.Source).ToArray();
         }
 
-        private static IEnumerable<string> ProbeRedirectionFiles(string docsetPath, Repository? repository)
+        private static IEnumerable<(string redirectionFile, bool IsSubRedirectionFile)> ProbeRedirectionFiles(
+            string docsetPath,
+            Repository? repository,
+            HashSet<string> redirectionFiles)
         {
-            yield return Path.Combine(docsetPath, "redirections.yml");
-            yield return Path.Combine(docsetPath, "redirections.json");
+            yield return (Path.Combine(docsetPath, "redirections.yml"), true);
+            yield return (Path.Combine(docsetPath, "redirections.json"), true);
 
             if (repository != null)
             {
-                yield return Path.Combine(repository.Path, ".openpublishing.redirection.json");
+                yield return (Path.Combine(repository.Path, ".openpublishing.redirection.json"), false);
+
+                // redirection files registered in .openpublishing.publish.config.json
+                foreach (var item in redirectionFiles)
+                {
+                    if (item.Equals(".openpublishing.redirection.json")
+                        || item.Equals(Path.GetRelativePath(repository.Path, Path.Combine(docsetPath, "redirections.yml")))
```

Comment:
`redirections.yml`/`redirections.json` is supposed to be under docset root. Do we have them under repo root as well?

Reply:
No repos use "redirection.yml" and "redirection.json" (only in spec tests).

Just to ignore them when `redirections.yml/redirections.json` is registered in ops.config. (Although impossible currently.)

Reply:
Then remove these equals check here

Reply:
🆗